### PR TITLE
Explicitly handle corner cases when inserting a  ROI in a TimeSplitter object

### PR DESCRIPTION
**Pull request by jmborr**
**Description of work**
Explicitly handle inserting an `ROI` that either ends at the beginning of the `TimeSplitter` or starts at the end of the `TimeSplitter`.  
Referenced by  [EWM Defect 1923 FilterEvents' TimeSplitterTest fails some of the time in the Windows builds](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1923).

**Purpose of work**
Bugfix  `TimeSplitterTest.to_splitters()`.

**To test:**
Just review the code.

*There is no associated issue.*

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: Framework/Kernel/src/SplittingInterval.cpp [hunk 1]
```diff
@@ -55,6 +55,13 @@ bool SplittingInterval::operator==(const SplittingInterval &ti) const {
   }
 }
 
+std::string SplittingInterval::debugStrPrint() const {
```
Comment (robertapplin) [inline Framework/Kernel/src/SplittingInterval.cpp:58]: Is it intended to keep this function in the code in case we need it in the future?

**Review by robertapplin — CHANGES_REQUESTED** [review]
Thank you for fixing this unreliable issue so quickly! I just had one question and then I'm happy to approve

File: Framework/Kernel/src/SplittingInterval.cpp (hunk 1 above)
Comment (jmborr) [reply Framework/Kernel/src/SplittingInterval.cpp:58]: yes

**Review by robertapplin — APPROVED** [review]
Thanks again for fixing this

**Review by thomashampson — DISMISSED** [review]
Would it be possible to add a unit test to reproduce this corner case?

**Comment by jmborr** [comment]
> Would it be possible to add a unit test to reproduce this corner case?

[There's already one test for this corner case](https://github.com/mantidproject/mantid/blob/addroi_corner_cases/Framework/DataObjects/test/TimeSplitterTest.h#L272)

**Comment by robertapplin** [comment]
> [There's already one test for this corner case](https://github.com/mantidproject/mantid/blob/addroi_corner_cases/Framework/DataObjects/test/TimeSplitterTest.h#L272)
> 
> The `test_to_splitters()` and this test would fail sporadically

Would having a test that fails 100% of the time (without this fix) be an easy addition, or is it not possible due to the nature of the bug?

**Comment by jmborr** [comment]
> > [There's already one test for this corner case](https://github.com/mantidproject/mantid/blob/addroi_corner_cases/Framework/DataObjects/test/TimeSplitterTest.h#L272)
> > The `test_to_splitters()` and this test would fail sporadically
> 
> Would having a test that fails 100% of the time (without this fix) be an easy addition, or is it not possible due to the nature of the bug?

as far as I can tell, this was a "sometimes fails" bug, not deterministic

**Comment by MialLewis** [comment]
@mantidproject/gatekeepers can we get this merged, the failed test is unrelated and it is causing the windows nightly pipeline to fall over on occasion?
